vendor/views/mixin.py

```diff
@@ -1,16 +1,39 @@
 from django.core.exceptions import ImproperlyConfigured
 from django.contrib import messages
-from django.utils.translation import ugettext as _
-from django.http import Http404
 from django.contrib.sites.models import Site
+from django.contrib.sites.shortcuts import get_current_site
+from django.http import Http404
 from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
 
 
-class ProductRequiredMixin():
+def get_site_from_request(request):
+    if hasattr(request, 'site'):
+        return request.site
+    return get_current_site(request)
+
+
+class SetSiteToRequestMixin:
+
+    def dispatch(self, request, *args, **kwargs):
+        request.site = get_site_from_request(request)
+        return super().dispatch(request, *args, **kwargs)
+
+
+class PassRequestToFormKwargsMixin:
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        self.request.site = get_site_from_request(self.request)
```

Comment:
This is probably not a safe assumption to do.  Vendor should not be modifying the request object.  Could lead to unintended issues.  If you need it in the context, add it to the context object instead.

Reply:
removed that line